### PR TITLE
Add CacheDev

### PR DESCRIPTION
**Pull request by mulkieran**
It's another type of device that we would like to support.

Signed-off-by: mulhern <amulhern@redhat.com>

### CODE REVIEW

**Comment by mulkieran** [comment]
Whoops.

**Comment by mulkieran** [comment]
Depends on #214.

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
Probably Jenkins failure is spurious.

**Comment by mulkieran** [comment]
@trgill Looks like Jenkins needs some cleanup.

**Comment by mulkieran** [comment]
@trgill Looks like I've broken Jenkins again.

**Comment by mulkieran** [comment]
@agrover @trgill review?

File: src/cachedev.rs [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+use super::result::{DmResult, DmError, ErrorEnum};
+use super::shared::{DmDevice, device_create, device_exists, device_setup};
+use super::types::{DataBlocks, DevId, DmName, DmUuid, MetaBlocks, Sectors, TargetLine,
+                   TargetTypeBuf};
+
+// Specified in kernel docs
+#[allow(dead_code)]
+const MIN_CACHE_BLOCK_SIZE: Sectors = Sectors(64); // 32 KiB
+#[allow(dead_code)]
+const MAX_CACHE_BLOCK_SIZE: Sectors = Sectors(2_097_152); // 1 GiB
```
Comment (agrover) [inline src/cachedev.rs]: These both are only used in the test code, should they be defined in mod tests?

We could use them for checking incoming params, but I think the kernel is already doing that.

Comment (mulkieran) [reply src/cachedev.rs]: Ok. I moved those and did similar for thinpooldev consts.

The kernel does do that checking, but it is, of course, extremely uninformative about what's wrong.
Nonetheless, I'm not terribly excited about doing more helpful checking our selves anytime soon. Once it giets out of sync with kernel's reality, thing would be quite grim.

**Comment by mulkieran** [comment]
I'm going to give a try at moving everything right into mod tests and see if ithat is nicer.

**Comment by mulkieran** [comment]
No. Sticking with this. I'm not happy with messing with publicity modifiers on test modules.

**Comment by mulkieran** [comment]
@trgill review?

**Comment by trgill** [comment]
I resubmitted the Jenkins test.  A previous run left one of the Jenkins nodes with:

# dmsetup ls
example-dev	(253:4)
cache	(253:3)
cache-origin	(253:2)
cache-meta	(253:0)
cache-cache	(253:1)

Just want to make sure that has been fixed.

**Pull request merged**